code/modules/shuttle/on_move.dm

```diff
@@ -396,10 +396,34 @@ All ShuttleMove procs go here
 
 	. = ..()
 
+/obj/docking_port/mobile/fob/onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, old_dock)
+	. = ..()
+
+	//Unload the planet
+	var/obj/docking_port/stationary/S = old_dock
+	for(var/datum/star_system/SYS in SSstarmap.star_systems)
+		var/datum/planet/PL = SYS.get_planet_for_z(oldT.z)
+		if(S.name == "FOB Landing Zone" && SSstarmap.current_planet.name != PL.name)
+			PL.no_unload_reason = ""
```

Comment:
Check name first before looping through every system in the game. 

Make it a typeless loop.

You don't need to check where the planet is. If the no unload reason was set by you on landing you must remove it when you leave.

Preferably put this in beforeShuttleMove

Reply:
You didn't address all the issues